File: 프로그래머스/1/42889. 실패율/실패율.py
def solution(N, stages):
    result = []
    for i in range(1,N+1):
        fail = 0
        challenge = 0
        
        for stage in stages:
            if stage >= i:
                if stage == i:
                    fail += 1
                challenge += 1
        if fail == 0 :
            result.append([0,-i])
        else:
            result.append([fail/challenge,-i])

    result.sort(reverse=True)
    ans = []
    for i in result:
        ans.append(i[1]*-1)
        
    return ans
        

File: 프로그래머스/1/42889. 실패율/실패율.js
// 해당 스테이지보다 큰 숫자는 참여자, 같은 숫자는 실행중인자 
// N개의 배열에서 [실패율,-번호] 순서대로 두기
// 내림차순 정렬

function solution(N, stages) {
    const play = new Array(N);
    const fail = new Array(N);
    play.fill(0)
    fail.fill(0)
    
    for (let i=0;i<N;i++){
        for (stage of stages){
            if (stage >= i+1){
                play[i] += 1
            }if (stage == i+1){
                fail[i] += 1
            }
        }
    }
    const arr = play.map((item,idx)=>{
        return [fail[idx]/item,idx+1]
    })
    arr.sort((a,b)=> b[0]-a[0])
    const answer = [];
    
    for (const i of arr){
        answer.push(i[1])
    }
    return answer
}